app/Api/page.tsx

```diff
@@ -0,0 +1,127 @@
+'use client'
+import Image from "next/image"
+import useState  from "react-usestateref";
+//import user from "../server/user.png";
+//import bot from "../server/bot.jpeg";
+
+enum Creator {
+    Me = 0,
+    Bot =1
+}
+
+interface MessageProps {
+    text : string;
+    from: Creator;
+    key: Number;
+
+}
+
+interface InputProps {
+    onSend: (input: string) => void;
+    disabled: boolean;
+
+}
+
+const ChatMessage = ({ text, from } : MessageProps) => {
+    return (
+        <>
+        {from == Creator.Me && (
+            <div className="bg-white p-4 rounded-lg flex gap-4 items-center whitespace-pre-wrap">
+                <p className="text-gray-700">{text}</p>
+
+            </div>
+        )}
+        {from == Creator.Bot && (
+            <div className="bg-white p-4 rounded-lg flex gap-4 items-center whitespace-pre-wrap">
+                <p className="text-gray-700">{text}</p>
+
+            </div>
+        ) }
+        </>
+    )
+}
+
+const ChatInput = ({ onSend, disabled}: InputProps) => {
+    const [input,setInput] = useState('');
+
+    const sendInput = () => {
+        onSend(input);
+        setInput('');
+    };
+
+    const handleKeyDown = (event: any) => {
+        if (event.keyCode === 13) {
+            sendInput();
+        }
+    };    
+    
+    return (
+        <div className=" bg-whiet border-2 p-2 rounded-lg flex justify-center">
+            <input
+            value= {input}
+            onChange={(ev: any) => setInput(ev.target.value)} 
+            className="w-full pu-2 px-3 text-gray-800 rounded-lg focus:outline-none"
+            type="text"placeholder= "Ask me anything"
+            disabled={disabled}
+            onKeyDown={(ev) => handleKeyDown(ev)} 
+            />
+        </div>
+    )
+
+}
+
+//our page
+export default function Chat() {
+    const[messages, setMessages, messageRef] = useState<MessageProps[]>([]);
+    const[loading, setLoading] = useState(false);
+
+    const callApi = async (input: string) => {
+        setLoading(true);
+        const myMessage: MessageProps = {
+            text: input,
+            from: Creator.Me,
+            key: new Date().getTime()
+        };
+
+        setMessages ([...messageRef.current, myMessage]);
+        const response = await fetch ('/api/generate-answer', {
```

Comment:
The callApi() functin is trying to send a request to /api/generate-answer, which is a URL that does not exist. This will result in an Error 404: Not Found. Instead, are you able to add in the ChatGPT features here?